BetterStacking/Patches.cs

```diff
@@ -1,145 +1,98 @@
-        internal class Panel_Inventory_Examine_SharpenSuccessful
-        {
-            private static void Prefix(Panel_Inventory_Examine __instance)
-            {
-                Implementation.SplitStack(__instance.m_GearItem);
-            }
-            private static void Postfix(Panel_Inventory_Examine __instance)
-            {
-                Implementation.AddToExistingStack(__instance.m_GearItem);
-            }
-        }*/
-
-        [HarmonyPatch(typeof(PlayerManager), nameof(PlayerManager.AddToExistingStackable), new System.Type[] { typeof(string), typeof(float), typeof(int), typeof(GearItem) })]
-        internal class PlayerManager_AddToExistingStackable
-        {
-            private static bool Prefix(PlayerManager __instance, ref GearItem __result, string itemName, float normalizedCondition, int numUnits, GearItem gearToAdd)
-            {
-                if (normalizedCondition == 0)
-                {
-                    __result = null;
-                    return false;
-                }
-
-                if (Implementation.UseDefaultStacking(gearToAdd))
-                {
-                    return true;
-                }
-
-                GearItem targetStack = GameManager.GetInventoryComponent().GetClosestMatchStackable(itemName, normalizedCondition);
-                if (!Implementation.CanBeMerged(targetStack, gearToAdd))
-                {
-                    __result = null;
-                    return false;
-                }
-
-                Implementation.MergeIntoStack(normalizedCondition, numUnits, targetStack);
-                __result = targetStack;
-                return false;
-            }
-        }
+    {
+
+        internal static bool postfix_track = false;
+        internal static float postfix_condition = 0;
+        internal static GearItem postfix_stack = new GearItem();
+        internal static GearItem postfix_geartoadd = new GearItem();
+        internal static float postfix_constraint = 0;
+
+        [HarmonyPatch]
+        internal static class PlayerManager_TryAddToExistingStackable
+        {
+            static MethodBase? TargetMethod()
+            {
+                MethodInfo[] methods = typeof(PlayerManager).GetMethods();
+                foreach (MethodInfo m in methods)
+                {
+                    if (m.Name == nameof(PlayerManager.TryAddToExistingStackable) && m.ReturnType == typeof(bool) && !m.IsGenericMethod && m.GetParameters().Length == 4)
+                    {
+                        return m;
+                    }
+                }
+                MelonLogger.Msg("PlayerManager.TryAddToExistingStackable not found for patch.");
+                return null;
+            }
+            internal static bool Prefix(ref GearItem gearToAdd, float normalizedCondition, int numUnits, out GearItem existingGearItem)
+            {
+
+                existingGearItem = new GearItem();

```

Comment:
As above, don't create new Unity components from managed code. Make it nullable.

Reply:
I'm fairly sure this line can be removed entirely.